docs/Developers/IO.rst

```diff
@@ -0,0 +1,35 @@
+coala's IO Mechanism
+====================
+
+The interface
```

Comment:
Maybe just "Interface"? :)

Reply:
Could be maybe emitted completely, since what follows is a general description of the IO features^^